src/main/java/org/subethamail/smtp/server/SessionHandler.java

```diff
@@ -0,0 +1,11 @@
+package org.subethamail.smtp.server;
+
+import org.subethamail.smtp.DropConnectionException;
+
+public interface SessionHandler {
+
+    public void acquire(Session session) throws DropConnectionException;
+
+    public void release(Session session);
```

Comment:
What about "onSessionEnd" ?

Reply:
I agree with @eolivelli on this in that an on- prefixed method name is better. This is a notification from the server that the session has been terminated. I prefer `onClose` or `onTerminate`.